### ISSUE TITLE
Using elemental with workflow applied to pages fails

### ISSUE THREAD

**Issue by phalkunz** [dnadesign/silverstripe-elemental#271]
When a workflow is applied to a page and the page can't be edited (e.g. because a workflow is active and awaiting approval) the elemental attempt to load the HTML content of each element into the CMS. This appears to be because of `ElementalArea->forTemplate()`.

This fails because in the CMS only the `cms-form` theme is active meaning the templates can't be found. A solution could be to just list the elements using the current editor but not allow them to be edited.

**Comment by madmatt** [comment]
Ah interesting - yeah that is exactly the same issue - although I suspect with a different outcome. I don't think that rendering the element as HTML inside the CMS is a good idea - as mentioned in #228 this would lead to super ganky CMS interfaces and could easily break the CMS too (with the right combination of HTML classes etc).

I think the right solution in this case would be to do a `performReadonlyTransformation()` or similar on the `ElementalEditor` field maybe - so that it lists the elements but doesn't allow them to be edited. As long as the CMS preview still loads that's the main thing for a simple workflow review IMO.

**Comment by robbieaverill** [comment]
Yeah, I totally agree that we shouldn't be rendering frontend HTML in the CMS.

The search issue at #228 renders it in order to use the DOM for searching, but never actually renders it. It's possible that this issue could be an advanced workflow issue rather than an elemental issue, but we can certainly investigate it anyway.

**Comment by NightJar** [comment]
@phalkunz can you please list the versions of things you have installed? I am having trouble replicating this.
`composer show` please :)

**Comment by phalkunz** [comment]
```
silverstripe/recipe-cms                     1.1.1
silverstripe/recipe-content-blocks          N/A
dnadesign/silverstripe-elemental            2.1.0
symbiote/silverstripe-advancedworkflow      5.0.6
```

I don't see `silverstripe/recipe-content-blocks `. Do you think that might be the problem?

**Comment by NightJar** [comment]
Thanks @phalkunz - `recipe-content-blocks` is essentially just a composer.json with a bundle of requirements related to `elemental` (and some extra things to pack out the block types beyond the default "content" block).

I'll test with the versions you've got installed and see if I can replicate again.
If not, I'll need to get a full report on steps to recreate.

**Comment by NightJar** [comment]
@phalkunz I'm still unable to reproduce this at those versions.

Steps I've taken:

composer.json:
```
    "require": {
        "php": ">=5.6.0",
        "silverstripe/recipe-plugin": "^1.2",
        "silverstripe/recipe-cms": "1.1.1",
        "symbiote/silverstripe-advancedworkflow": "5.0.6",
        "dnadesign/silverstripe-elemental": "2.1.0"
    },
```
Composer install
Copy the blocks page from https://github.com/silverstripe/recipe-content-blocks/blob/381b69cbc1e71c3990117b519bf983d14681ba89/app/src/BlocksPage.php into `mysite/code/BlocksPage.php`
run a dev/build
Create a workflow, titled "default" with the 2 step approval template, and the button text "Go".
Create a Blocks Page in the CMS
Go to the settings tab for that new page and apply the `default` workflow (created above)
Publish the BlocksPage
Create a user in the Security Tab, assigning it the Content Editors Group
log in on a different browser to the CMS
Visit the Blocks Page, edit the content, and click the "Go" button.
Refresh the CMS
Observe a read only view of the Edit Form for the Blocks Page.

**Comment by robbieaverill** [comment]
@NightJar do you have a custom block type with its template living in the theme rather than in a module? I think that's a pretty important part of this issue

**Comment by madmatt** [comment]
Yeah it seems to occur when there's a template for a block that's in a theme... but I can't see why (in @NightJar's screenshot it's clearly just showing all of the fields of that block as read-only - not trying to render the frontend interface)... maybe there's a fallback or something in place.

**Comment by NightJar** [comment]
haha, of course :sweat: 
Will try again.

**Comment by brynwhyman** [comment]
@NightJar @robbieaverill in the meantime can you please give detail of the potential workaround for this bug

**Comment by robbieaverill** [comment]
Workaround should be the same as #228 which is to make a copy of any custom block templates and put them into your `app` or `mysite` templates folder, so that there is a template available outside of the themes (which aren't used in the CMS).
